File: app/src/main/java/com/example/kursach/domain/model/UserProfile.kt
package com.example.kursach.domain.model

import java.time.Instant
import java.time.LocalDate

data class UserProfile(
    val id: Int,
    val email: String,
    val firstName: String,
    val lastName: String,
    val birthDate: LocalDate,
    val region: String,
    val city: String,
    val createdAt: Instant,
    val updatedAt: Instant,
) {
    val fullName: String
        get() = "$firstName $lastName"
}

File: lib/src/domain/entities/user_profile.dart
class UserProfile {
  const UserProfile({
    required this.id,
    required this.email,
    required this.firstName,
    required this.lastName,
    required this.birthDate,
    required this.createdAt,
    required this.updatedAt,
  });

  final int id;
  final String email;
  final String firstName;
  final String lastName;
  final DateTime birthDate;
  final DateTime createdAt;
  final DateTime updatedAt;

  String get fullName => '$firstName $lastName';
}
